components/region_cache_memory_engine/src/range_manager.rs

```diff
@@ -276,4 +333,35 @@ mod tests {
         assert!(!range_mgr.evict_range(&r_right));
         assert!(range_mgr.historical_ranges.get(&r_right).is_none());
     }
+
+    #[test]
+    fn test_range_load() {
+        let mut range_mgr = RangeManager::default();
+        let r1 = CacheRange::new(b"k00".to_vec(), b"k10".to_vec());
+        let r2 = CacheRange::new(b"k10".to_vec(), b"k20".to_vec());
+        let r3 = CacheRange::new(b"k20".to_vec(), b"k30".to_vec());
+        let r4 = CacheRange::new(b"k25".to_vec(), b"k35".to_vec());
+        range_mgr.new_range(r1.clone());
+        range_mgr.new_range(r3.clone());
+        range_mgr.evict_range(&r1);
+
+        let mut gced = BTreeSet::default();
+        gced.insert(r2.clone());
+        range_mgr.set_ranges_in_gc(gced);
```

Comment:
since loading and gc are both sync and run in the same worker, can this happen? see my earlier comment in background.rs

Reply:
Load is async. It first push the range in the pending_ranges in the RangeManager, then a LoadTask can be scheduled to the worker.

Reply:
You mean for the same range, load and gc performs concurrently?

Reply:
Actually, set_ranges_in_gc is to avoid the range being gced and loaded simutaneously.